homeassistant/components/websocket_api/decorators.py

```diff
@@ -56,34 +64,32 @@ def with_admin(hass, connection, msg):
 
 
 def ws_require_user(
-    only_owner=False,
-    only_system_user=False,
-    allow_system_user=True,
-    only_active_user=True,
-    only_inactive_user=False,
-):
+    only_owner: bool = False,
+    only_system_user: bool = False,
+    allow_system_user: bool = True,
+    only_active_user: bool = True,
+    only_inactive_user: bool = False,
+) -> Callable[[const.WebSocketCommandHandler], const.WebSocketCommandHandler]:
     """Decorate function validating login user exist in current WS connection.
 
     Will write out error message if not authenticated.
     """
 
-    def validator(func):
+    def validator(func: const.WebSocketCommandHandler) -> const.WebSocketCommandHandler:
         """Decorate func."""
 
         @wraps(func)
-        def check_current_user(hass, connection, msg):
+        def check_current_user(
+            hass: HomeAssistant, connection: ActiveConnection, msg: dict[str, Any]
+        ) -> None:
             """Check current user."""
 
-            def output_error(message_id, message):
+            def output_error(message_id: str, message: str) -> None:
                 """Output error message."""
                 connection.send_message(
                     messages.error_message(msg["id"], message_id, message)
                 )
 
-            if connection.user is None:
```

Comment:
`connection.user` is set from `RefreshToken.user`. I checked both places where `RefreshToken` is created in `homeassistant/auth/auth_store.py` and they both look to me as not `None`.

And `RefreshToken.user` is typed accordingly as not optional.

Reply:
@balloob can you verify this?

Reply:
Correct. All refresh tokens are bound to a user (which can be "system generated").

Reply:
Just rememebred. This is a relic of the past. We used to have `api_password` logins which were not bound to a user.